### ISSUE TITLE
Hasselblad H4D-50

### ISSUE THREAD

**Issue by eyedear** [darktable-org/darktable#5305]
hi I have uploaded a Hasselblad H4D-50 RAW file to raw.pixls.us I hope darktable can support this camera RAW. if you need any extra images please let me know. I can upload more. if there is any instruction that I can help in making this easier let me know 

I also manage to find a color map files if needed please let me know

Before filing any support requests, please do make sure that you have already
contributed the whole raw sample set for your camera under CC0 license to
https://raw.pixls.us/ !

#### Important note: to make sure that darktable can keep on supporting the raw
file format for your camera, *please* read
[this post](https://discuss.pixls.us/t/raw-samples-wanted/5420?u=lebedevri) on
how/what raw samples you can contribute to ensure that we have the *full* raw
sample set for your camera under CC0 license!

**Comment by LebedevRI** [comment]
> hi I have uploaded a Hasselblad H4D-50 RAW file to raw.pixls.us

Thank you for the contribution!

**Comment by eyedear** [comment]
@LebedevRI it should be similar to the Hasselblad CFV-50 currently supported on rawspeed camera support page which uses the same sensor. hope this information helps.

**Comment by eyedear** [comment]
I added the following entry into the cameras.xml file

<Camera make="Hasselblad" model="Hasselblad H4D-50">
		<ID make="Hasselblad" model="H4D-50">Hasselblad 50-Coated</ID>
		<CFA width="2" height="2">
			<Color x="0" y="0">RED</Color>
			<Color x="1" y="0">GREEN</Color>
			<Color x="0" y="1">GREEN</Color>
			<Color x="1" y="1">BLUE</Color>
		</CFA>
		<Crop x="48" y="6" width="-42" height="-82"/>
		<Sensor black="256" white="65535"/>
	</Camera>

the files can be open in darkroom but only on MACOS in windows some of the fff files can be open but many cannot.
once open in darkroom an error show "hasselblad H4D-50 color matrix not found!"

I am attaching screenshots I hope this make it easier for you @LebedevRI 
![Screenshot 2020-06-05 at 5 19 43 PM](https://user-images.githubusercontent.com/7136598/83860296-acb5b300-a751-11ea-907c-b4d22d5d051f.jpg)
![Screenshot 2020-06-05 at 5 19 52 PM](https://user-images.githubusercontent.com/7136598/83860304-af180d00-a751-11ea-92af-6eca14422faf.jpg)

**Comment by eyedear** [comment]
i have added this line to the  adobe-coeff.c file

{ "Hasselblad H4D-50", { 5656,-659,-346,-3923,12306,1791,-1602,3509,5442 } },

I have compile it on windows and it works. 
another issue i am having is that the white point of the raw file is different at different ISO as below

iso 50 = 65535
iso 100 = 32767
iso 200 = 16383
iso 400 = 8191
iso 800 = 4095

the value when input into the RAW black/White Point modules works though i still need to create a custom base curves to get the contrast correct. 

hope someone can test this and if it works add this in to the source as i don't know how

**Comment by LebedevRI** [comment]
> another issue i am having is that the white point of the raw file is different at different ISO as below

Looking at exiv2/exiftool output for `.fff`, looks like the info is present
```
Exif.SubImage1.BlackLevel                     256/1
Exif.SubImage1.WhiteLevel                     32767
```
so we likely should just read that [here](https://github.com/darktable-org/rawspeed/blob/708f3cdc7bf1e1a7cbe52c479a625bda6d08455c/src/librawspeed/decoders/ThreefrDecoder.cpp#L75), like we do for [some other formats](https://github.com/darktable-org/rawspeed/blob/708f3cdc7bf1e1a7cbe52c479a625bda6d08455c/src/librawspeed/decoders/PefDecoder.cpp#L113-L120).

**Comment by eyedear** [comment]
yes the info is present but it does not affect the black/WhitePoint module.it will follow the cameras.xml setting. I can only report as I am not that familiar with the code. 
I have use the color matrix from the cfv-50 and it works as I remember they have the same sensor. in the .fff file there is actually a colormatrix2 entry but once converted to DNG it disappear  and that the issue
